src/enum.js

```diff
@@ -26,8 +26,13 @@ export class Enum extends XdrPrimitiveType {
    * @inheritDoc
    */
   static write(value, writer) {
-    if (!(value instanceof this))
-      throw new XdrWriterError(`unknown ${value} is not a ${this.enumName}`);
+    if (!this.isValid(value)) {
```

Comment:
I'm assuming that `this` is the Class, but personally I find it less confusing if you use the Class directly to remove any ambiguities. Plus the caller could have used `bind` to change this method's `this`. Not that we need to worry, but it's possible.
```suggestion
    if (!Enum.isValid(value)) {
```

Reply:
Unfortunately, we have to use `this` here for an oddly-specific reason: we need to make sure it deals with child classes and any modifications that are made after creation, e.g. like [here](https://github.com/stellar/js-xdr/blob/master/src/union.js#L99-L102). Just passing the class name doesn't cover this case.

Reply:
But shouldn't the classed be sealed? Why do we need to consider the case of subclasses?

Reply:
You can see in that linked line that a `Union` itself is never actually created, it's always a locally-generated subclass of `ChildUnion` that has a bunch of properties attached to it (most importantly `unionName`) that gets returned. So we'd need to be able to `[the specific ChildUnion class].isValid` if we wanted an accurate evaluation of "this"ness.